kong/db/dao/server_names.lua

```diff
@@ -0,0 +1,159 @@
+local singletons = require "kong.singletons"
+
+-- A "name_list" is an array of server names like { "example.com", "foo.com" }
+-- All methods containing "list" on their name act on name lists, not on lists
+-- of complete ServerName entities
+
+-- { "a", "b", "c" } - { "a", "c" } = { "b" }
+local function list_diff(list1, list2)
+  local set2 = {}
+  for i=1, #list2 do
+    set2[list2[i]] = true
+  end
+
+  local diff = {}
+  local len = 0
+  for i=1, #list1 do
+    if not set2[list1[i]] then
+      len = len + 1
+      diff[len] = list1[i]
+    end
+  end
+
+  return diff
+end
+
+
+local _ServerNames = {}
+
+
+-- Truthy if all the names on the list don't exist on the db
+function _ServerNames:check_list_is_new(name_list)
+  -- when creating a new cert (no cert_id provided):
+  -- dont add the certificate or any names if we have an server name conflict
+  -- its fairly inefficient that we have to loop twice over the datastore
+  -- but no support for OR queries means we gotsta!
+  for i=1, #name_list do
+    local name = name_list[i]
+    local row, err, err_t = singletons.db.server_names:select_by_name(name)
+    if err then
+      return nil, err, err_t
+    end
+
+    if row then
+      -- Note: it could be that the name is not associated with any
+      -- certificate, but we don't handle this case. (for PostgreSQL
+      -- only, as C* requires a cert_id for its partition key).
+      local msg   = "Server name already exists: " .. name
+      local err_t = self.errors:conflicting_input(msg)
+      return nil, tostring(err_t), err_t
+    end
+  end
+
+  return 1
+end
+
+
+-- Truthy if all the names on the list don't exist on the db or exist but are
+-- associated to the given certificate
+function _ServerNames:check_list_is_new_or_in_cert(cert_pk, name_list)
+  for i=1, #name_list do
+    local row, err, err_t = self:select_by_name(name_list[i])
+    if err then
+      return nil, err, err_t
+    end
+    if row and row.certificate.id ~= cert_pk.id then
+      local msg = "Server Name '" .. row.name ..
```

Comment:
On line 47 it says `"Server name already exists` and here `Server Name `. I think in general these messages should be all lower case (because they are injected usually in different places), comments on that?